src/main/java/com/seedfinding/latticg/math/component/BigVector.java

```diff
@@ -445,18 +446,17 @@ public String toString() {
         StringBuilder sb = new StringBuilder("{");
 
         for (int i = 0; i < this.getDimension(); i++) {
-            sb.append(this.get(i))
-                    .append(i == this.getDimension() - 1 ? "" : ", ");
+            sb.append(this.get(i)).append(i == this.getDimension() - 1 ? "" : ", ");
         }
 
         return sb.append("}").toString();
     }
 
-    public String toApproximatetring() {
+    public String toApproximateString() {
         StringBuilder sb = new StringBuilder("[");
 
         for (int i = 0; i < this.getDimension(); i++) {
-            sb.append(this.get(i).toDouble()).append(i == this.getDimension() - 1 ? "" : " ");
+            sb.append(this.get(i).toBigDecimal(MathContext.UNLIMITED)).append(i == this.getDimension() - 1 ? "" : " ");
```

Comment:
Doesn't this have issues if the fraction is not exactly representable as a decimal?